### PR TITLE
rac2: add priority conversion functions

### PR DESCRIPTION
**Pull request by sumeerbhola**
For converting between rafpb.Priority and admissionpb.WorkPriority.

Informs #128309

Epic: CRDB-37515

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/kvflowcontrol/rac2/priority.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func AdmissionPriorityToRaftPriority(pri admissionpb.WorkPriority) raftpb.Priority {
+	if pri < admissionpb.NormalPri {
+		return raftpb.LowPri
+	} else if pri < admissionpb.LockingNormalPri {
+		return raftpb.NormalPri
+	} else if pri < admissionpb.UserHighPri {
+		return raftpb.AboveNormalPri
+	} else if pri <= admissionpb.HighPri {
+		return raftpb.HighPri
+	} else {
```
Comment (pav-kv) [inline pkg/kv/kvserver/kvflowcontrol/rac2/priority.go:29]: Do you know the most frequent one(s) here, for fast-path? Could place it first, and/or do something binary-search-like to reduce the max number of comparisons:

```golang
if pri < LockingNormalPri {
	if pri < NormalPri {
		return LowPri
	}
	return NormalPri
}
if pri < UserHighPri {
	return AboveNormalPri
}
return HighPri
```

Same below.

File: pkg/kv/kvserver/kvflowcontrol/rac2/priority.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+package rac2
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/raft/raftpb"
+	"github.com/cockroachdb/cockroach/pkg/util/admission/admissionpb"
+)
+
+// AdmissionPriorityToRaftPriority maps the larger set of values in
+// admissionpb.WorkPriority to the smaller set of raftpb.Priority.
+func AdmissionPriorityToRaftPriority(pri admissionpb.WorkPriority) raftpb.Priority {
```
Comment (pav-kv) [inline pkg/kv/kvserver/kvflowcontrol/rac2/priority.go:20]: nit: shorten the names? E.g. `ToRaftPriority / ToAdmissionPriority` or `AdmissionToRaftPriority / RaftToAdmissionPriority`.

**Review by pav-kv — APPROVED** [review]

**Review by kvoli — APPROVED** [review]

**Review by sumeerbhola — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/128721#-:-O45rdaJ1GDdbHkeQj4u:b-814g8k)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @pav-kv)

___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/priority.go` line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/128721#gh-1713366384:-O45o7paDdeUMNHW3gOD:bs7wumx) ([raw file](https://github.com/cockroachdb/cockroach/blob/bdb6d9a9ad0230bf36e3fec2d5ae3b3b06eae4d8/pkg/kv/kvserver/kvflowcontrol/rac2/priority.go#L20)):*
<details><summary><i>Previously, pav-kv (Pavel Kalinnikov) wrote…</i></summary><blockquote>

nit: shorten the names? E.g. `ToRaftPriority / ToAdmissionPriority` or `AdmissionToRaftPriority / RaftToAdmissionPriority`.
</blockquote></details>

Done. 
___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/priority.go` line 29 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/128721#gh-1713361568:-O45qqho-Z9ZJXQgpT83:b-k7060z) ([raw file](https://github.com/cockroachdb/cockroach/blob/bdb6d9a9ad0230bf36e3fec2d5ae3b3b06eae4d8/pkg/kv/kvserver/kvflowcontrol/rac2/priority.go#L29)):*
<details><summary><i>Previously, pav-kv (Pavel Kalinnikov) wrote…</i></summary><blockquote>

Do you know the most frequent one(s) here, for fast-path? Could place it first, and/or do something binary-search-like to reduce the max number of comparisons:

```golang
if pri < LockingNormalPri {
	if pri < NormalPri {
		return LowPri
	}
	return NormalPri
}
if pri < UserHighPri {
	return AboveNormalPri
}
return HighPri
```

Same below.
</blockquote></details>

I think it matters when comparisons are expensive, such as string comparisons. I wrote a small benchmark and for every input priority on may macbook pro get ~0.315 ns/op.
In this case NormalPri will be the common case, and it requires two comparisons regardless of what we do. Same for `RaftToAdmissionPriority`.

**Review by pav-kv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/128721)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained

___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/priority.go` line 29 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/128721#gh-1713361568:-O45sSBn8nqtyHvsE0C8:b4c9lyr) ([raw file](https://github.com/cockroachdb/cockroach/blob/bdb6d9a9ad0230bf36e3fec2d5ae3b3b06eae4d8/pkg/kv/kvserver/kvflowcontrol/rac2/priority.go#L29)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

I think it matters when comparisons are expensive, such as string comparisons. I wrote a small benchmark and for every input priority on may macbook pro get ~0.315 ns/op.
In this case NormalPri will be the common case, and it requires two comparisons regardless of what we do. Same for `RaftToAdmissionPriority`.
</blockquote></details>

:thumbsup:

**Comment by sumeerbhola** [comment]
bors r=pav-kv,kvoli

**Pull request merged**